File: Exercise/Hard/12. Centrifuge Balance Checker/Code/Centrifuge Balance Checker.cpp
#include <iostream>
#include <sstream>
#include <string>

using namespace std;

bool cFuge(int n, int k) {
    if (n == 1 || k == 1) {
        return false;
    }
    if (n % 2 != 0) {
        if (n == k) {
            return true;
        } 
        else {
            return false;
        }
    } 
    else if ((n - k) % 2 == 0) {
        return true;
    }
    return false; 
}

int main() {
    string input;
    getline(cin, input);

    size_t commaPos = input.find(',');
        if (commaPos != string::npos) {
        string nStr = input.substr(0, commaPos);
        string kStr = input.substr(commaPos + 1);
        
        int n, k;
        n = stoi(nStr);
        k = stoi(kStr);

        if (cFuge(n, k)) {
            cout << "true" << endl;
        } 
        else {
            cout << "false" << endl;
        }
    }

    return 0;
}

File: Exercise/Hard/12. Centrifuge Balance Checker/Code/Centrifuge Balance Checker.py
def cFuge(n, k):
    if n == 1 or k == 1:
        return False
    if n % 2 != 0:
        return n == k
    return (n - k) % 2 == 0

input_str = input()
n_str, k_str = input_str.split(',')

n = int(n_str)
k = int(k_str)

if cFuge(n, k):
    print("true")
else:
    print("false")

